### ISSUE TITLE
No css calsses are applied to prod version after `stencil push`

### ISSUE THREAD

**Issue by kstulgys** [TrellisCommerce/bigcommerce-cornerstone-tailwind-starter-base#9]
### Expected behavior
css calsses are applied to prod version after `stencil push`
### Actual behavior
No css calsses are applied to prod version after `stencil push`
### Steps to reproduce behavior
1. `stencil push`
2. go to bigcommerce admin
3. apply theme
4. go to store website

N.B. locally thing worky as expected

**Comment by travish** [comment]
Steps to reproduce:

**Preconditions**

1. `git checkout -f main`
2. `nvm use 18`
3. `npm install`
4. `npm run tailwind`

**Steps**

1. Open `templates/pages/account/orders/all.html`, and add a `tw-uppercase` to the `h1`, alongside the `.page-heading` class.
2. Open `config.json` and give my theme a unique name (for easy identification).
3. Test with `stencil bundle`
4. Push with `yes | stencil push -d`

**Expected Result**

CSS classes are applied to `production` after `stencil push`

**Actual Result**

CSS classes are applied to `production` after `stencil push`

https://share.trellis.co/Blub80EW

Sorry @kstulgys, I'm not able to reproduce your reported issue. Can you clarify (or create a screencast) so I can gather more context?

Thanks!

Edit: Fixed filename typo: `config.php` -> `config.json`

**Comment by kstulgys** [comment]
Thank you for the response. I can confirm that it's working. Although some things are unclear.
- do I need to run `stencil bundle` every time before `stencil push` ?
- where is config.php ?
- with tailwind base config it does't work in prod (the original issue):

```
module.exports = {
  content: ["./templates/**/*.{html,js}"],
  theme: {
    extend: {},
  },
  plugins: [],
}
```

**Comment by travish** [comment]
@kstulgys,

Thanks for your continued prompt responses.

1. `stencil bundle` isn't necessary - I run it to verify the files before I `stencil push`.
2. This is my fault, a typo. I meant `config.json`. Good catch!
3. I can reproduce this issue. It tracks to the pseudo-elements being generated in `assets/scss/tailwind.scss`, which throws a 500 error on the file in BigCommerce. Note, this issue DOES NOT happen locally with `stencil start`.

If you set `optimizeUniversalDefaults: true`, the issue is solved.

```javascript
module.exports = {
  content: ['./templates/**/*.html'],
  theme: {
    extend: {},
  },
  plugins: [],
  experimental: {
    optimizeUniversalDefaults: true,
  },
};
```

Let me know if you have success, I'll keep this issue open until then!

**Comment by kstulgys** [comment]
Something has todo with that theme object. I noticed that if it's:
```
 theme: {
    extend: {},
  },
```
Then it works locally, but in prod there are no styles.

**Comment by travish** [comment]
Correct.

As noted in my comment above, it has to do with the way BigCommerce is parsing the CSS file, which errors, and throws a 500 HTTP status. I found the same result as you locally - everything works as expected.

While it may have something to do with the theme object, I did not dig deeper. I'm content with the solution here to use `optimizeUniversalDefaults: true`.

Closing this issue. Thanks for your continued engagement!
